oioioi/mp/controllers.py

```diff
@@ -0,0 +1,198 @@
+from django.shortcuts import redirect
+from django.template.loader import render_to_string
+from django.template.response import TemplateResponse
+from django.utils.translation import gettext_lazy as _
+
+
+from oioioi.base.utils.query_helpers import Q_always_true
+from oioioi.base.utils.redirect import safe_redirect
+from oioioi.contests.models import Submission
+from oioioi.mp.models import MPRegistration, SubmissionScoreMultiplier
+from oioioi.mp.score import FloatScore
+from oioioi.participants.controllers import ParticipantsController
+from oioioi.participants.models import Participant
+from oioioi.participants.utils import is_participant
+from oioioi.programs.controllers import ProgrammingContestController
+from oioioi.rankings.controllers import DefaultRankingController
+
+CONTEST_RANKING_KEY = 'c'
+
+
+class MPRegistrationController(ParticipantsController):
```

Comment:
I feel that some methods here are overridden just like in other similar apps. Should we restructure the code to use some Mixins for avoiding to repeat same code over and over again?